SU2_CFD/src/iteration/CFluidIteration.cpp

```diff
@@ -126,6 +126,15 @@ void CFluidIteration::Iterate(COutput* output, CIntegration**** integration, CGe
     config[val_iZone]->SetGlobalParam(RANS, RUNTIME_SCALAR_SYS);
     integration[val_iZone][val_iInst][SCALAR_SOL]->MultiGrid_Iteration(geometry, solver, numerics, config, 
                                                                        RUNTIME_SCALAR_SYS, val_iZone, val_iInst);
+
+    // In case of turbulence, the Turb-Post computes the correct eddy viscosity based on mixture-density and
+    // mixture lam-visc. In order to get the correct mixture properties, based on the just updated mass-fractions, the
+    // Flow-Pre has to be called upfront. The updated eddy-visc are copied into the flow-solver Primitive in another 
+    // Flow-Pre call which is done at the start of the next iteration.
+    if(config[val_iZone]->GetKind_Turb_Model()) {
+      solver[val_iZone][val_iInst][MESH_0][FLOW_SOL]->Preprocessing(geometry[val_iZone][val_iInst][MESH_0], solver[val_iZone][val_iInst][MESH_0], config[val_iZone], MESH_0, NO_RK_ITER, RUNTIME_FLOW_SYS, false);
+      solver[val_iZone][val_iInst][MESH_0][TURB_SOL]->Postprocessing(geometry[val_iZone][val_iInst][MESH_0], solver[val_iZone][val_iInst][MESH_0], config[val_iZone], MESH_0);
+    }
```

Comment:
In the end the restart was fine but the main-iteration was missing this update of eddy viscosity to be correct (or consistent with the restart).